tests/components/homewizard/test_init.py

```diff
@@ -171,3 +174,92 @@ async def test_load_handles_initialization_error(aioclient_mock, hass):
     await hass.async_block_till_done()
 
     assert entry.state is ConfigEntryState.SETUP_RETRY or ConfigEntryState.SETUP_ERROR
+
+
+async def test_coordinator_sets_update_interval(aioclient_mock, hass):
+    """Test coordinator calculates correct update interval."""
+
+    device = get_mock_device()
+
+    entry = MockConfigEntry(
+        domain=DOMAIN,
+        data={CONF_IP_ADDRESS: "1.1.1.1"},
+        unique_id=DOMAIN,
+    )
+    entry.add_to_hass(hass)
+
+    with patch(
+        "aiohwenergy.HomeWizardEnergy",
+        return_value=device,
+    ):
+        await hass.config_entries.async_setup(entry.entry_id)
+
+    await hass.async_block_till_done()
+
+    assert hass.data[DOMAIN][entry.entry_id][
+        "coordinator"
+    ].update_interval == timedelta(seconds=5)
+
+
+async def test_coordinator_failed_to_update(aioclient_mock, hass):
+    """Test coordinator handles failed update correctly."""
+
+    # Update failed by internal error
+    async def _failed_update() -> bool:
+        return False
+
+    device = get_mock_device()
+
+    entry = MockConfigEntry(
+        domain=DOMAIN,
+        data={CONF_IP_ADDRESS: "1.1.1.1"},
+        unique_id=DOMAIN,
+    )
+    entry.add_to_hass(hass)
+
+    with patch(
+        "aiohwenergy.HomeWizardEnergy",
+        return_value=device,
+    ):
+        await hass.config_entries.async_setup(entry.entry_id)
+
+    await hass.async_block_till_done()
+
+    assert entry.state is ConfigEntryState.LOADED
+
+    device.update = _failed_update
+
+    with raises(UpdateFailed):
+        await hass.data[DOMAIN][entry.entry_id]["coordinator"]._async_update_data()
```

Comment:
Move time forward instead and assert the state of an entity. It should be unavailable.